homeassistant/components/remote/xiaomi_miio.py

```diff
@@ -0,0 +1,255 @@
+        """Initialize the remote."""
+        self._name = friendly_name
+        self._device = device
+        self._is_hidden = hidden
+        self._slot = slot
+        self._timeout = timeout
+        self._state = False
+        self._commands = commands
+
+    @property
+    def name(self):
+        """Return the name of the remote."""
+        return self._name
+
+    @property
+    def device(self):
+        """Return the remote object."""
+        return self._device
+
+    @property
+    def hidden(self):
+        """Return if we should hide entity."""
+        return self._is_hidden
+
+    @property
+    def slot(self):
+        """Return the slot to save learned command."""
+        return self._slot
+
+    @property
+    def timeout(self):
+        """Return the timeout for learning command."""
+        return self._timeout
+
+    @property
+    def is_on(self):
+        """Return False if device is unreachable, else True."""
+        from miio import DeviceException
+        try:
+            self.device.info()
+            return True
+        except DeviceException:
+            return False
+
+    @property
+    def should_poll(self):
+        """We should not be polled for device up state."""
+        return False
+
+    @property
+    def device_state_attributes(self):
+        """Hide remote by default."""
+        if self._is_hidden:
+            return {'hidden': 'true'}
+        else:
+            return
+
+    # pylint: disable=R0201
+    @asyncio.coroutine
+    def async_turn_on(self, **kwargs):
+        """Turn the device on."""
+        _LOGGER.error("Device does not support turn_on, " +
+                      "please use 'remote.send_command' to send commands.")
+
+    @asyncio.coroutine
+    def async_turn_off(self, **kwargs):
+        """Turn the device off."""
+        _LOGGER.error("Device does not support turn_off, " +
+                      "please use 'remote.send_command' to send commands.")
+
+    # pylint: enable=R0201
+    def _send_command(self, payload):
+        """Send a command."""
+        from miio import DeviceException
+
+        _LOGGER.debug("Sending payload: '%s'", payload)
+        try:
+            self.device.play(payload)
+        except DeviceException as ex:
+            _LOGGER.error(
+                "Transmit of IR command failed, %s, exception: %s",
+                payload, ex)
+
+    def send_command(self, command, **kwargs):
+        """Wrapper for _send_command."""
+        num_repeats = kwargs.get(ATTR_NUM_REPEATS)
+
+        delay = kwargs.get(ATTR_DELAY_SECS, DEFAULT_DELAY_SECS)
+
+        for _ in range(num_repeats):
+            for payload in command:
+                if payload in self._commands:
+                    for local_payload in self._commands[payload][CONF_COMMAND]:
+                        self._send_command(local_payload)
+                else:
+                    self._send_command(payload)
+                time.sleep(delay)
```

Comment:
If the xiaomi_miio library supports asyncio it was better when this method was async, since we want to sleep here. We usually frown upon sync sleep in worker threads which this now does. On the other hand async sleep doesn't cost anything.

I'm not sure why @pvizeli wanted this to be done sync?